### DISCUSSION TITLE
Not all subnets working with Cisco Unity extension

### DISCUSSION THREAD

**Discussion by eppesuig** [strongswan/strongswan#263] [Q&A]
Hello,
I am connecting a Linux server with an unknown partner, and I have to offer routes for 2 subnets, and to get routes for 4 different subnets. The remote peer require IKEv1, so I had to enable the Cisco Unity extension in order to have many subnets.

Now, the connection works but only two of the four remote routes are set up. In fact, the command "ip xfrm policy list" only lists 6 policies (3 for each of the remote 2 subnets). Sometimes I see 9 policies, where the 3rd group is about routing between a remote subnet and the second local subnet.

Logs do not contains errors, but routing is not working as expected. What kind of checks should I do? Is there any known problem with 4 routes and the Cisco Unity? Is there any problem in have a different number of routes as client and as a server? The xfrm policy is created when the connection is completed or only when the traffic really flows between the subnets?

The problem happens on a Debian 8.11 machine, with linux kernel 3.16, strongswan 5.5.1, iproute2 3.16

The connection in ipsec.conf is about this one (only public IP have been changed)

```
conn central
  # l = left = local
  # r = right = remote
  left=**LocalPublicIP**
  leftid=**LocalPublicIP**
  leftsubnet=192.168.10.0/24,192.168.11.0/24
  right=**RemotePublicIP**
  rightid=**RemotePublicIP**
  rightsubnet=10.163.4.0/22,10.143.12.0/22,10.178.4.0/22,10.172.4.0/22
  aggressive=no
  esp=3des-sha1-modp1536
  ike=aes-sha1-modp1536
  rightauth=psk
  leftauth=psk
  ikelifetime=60m
  keylife=20m
  rekeymargin=3m
  keyingtries=0
  keyexchange=ike
  auto=add
```

Thank you very much

**Comment by Thermi** [comment]
> The remote peer require IKEv1, so I had to enable the Cisco Unity extension in order to have many subnets.

No, just configure each combination of subnet as a seperate conn, as shown on the [wiki](https://wiki.strongswan.org/projects/strongswan/wiki/UsableExamples#Site-To-Site-Scenario).

Then it works fine.

Unity mode is for really old IKEv1 roadwarriors where enumeration of each subnet combination is not always possible - or the CISCO software just can't properly handle several CHILD_SAs per IKE_SA for roadwarriors.

**Reply by eppesuig** [reply]
Hello @Thermi,
thank you for your answer. I read the page you pointed out and it seems to me that it only apply when on your site you have to export many subnets. In my case I should export 2 subnets and import 4 remote subnets, as well.

Should I create one base connection and 2x4 derived connections, one for each pair local/remote subnet? Or, should I create 2 derived connections, one for each of my local subnets, each of them having all remote subnets as in `rightsubnet=10.163.4.0/22,10.143.12.0/22,10.178.4.0/22,10.172.4.0/22`.

Thank you,
Giuseppe

**Comment by Thermi** [comment]
Ikev1 only supports one subnet per side per CHILD_SA, so you have to enumerate them all (2x4).

Am March 16, 2021 8:19:07 AM UTC schrieb eppesuig ***@***.***>:
>Hello @Thermi,
>thank you for your answer. I read the page you pointed out and it seems
>to me that it only apply when on your site you have to export many
>subnets. In my case I should export 2 subnets and import 4 remote
>subnets, as well.
>
>Should I create one base connection and 2x4 derived connections, one
>for each pair local/remote subnet? Or, should I create 2 derived
>connections, one for each of my local subnets, each of them having all
>remote subnets as in
>`rightsubnet=10.163.4.0/22,10.143.12.0/22,10.178.4.0/22,10.172.4.0/22`.
>
>Thank you,
>Giuseppe
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/263#discussioncomment-487446

-- 
Sent from mobile
